File: apps/web/src/lib/types/sessions.ts
/**
 * Frontend types for chat sessions and messages (aligned with backend).
 */

export interface SessionSummary {
  id: string;
  title: string;
  created_at: string;
  updated_at: string;
  message_count: number;
}

export interface ChatMessageRecord {
  id: string;
  session_id: string;
  role: string;
  content: string;
  sources?: unknown[];
  debug?: Record<string, unknown> | null;
  error?: string | null;
  created_at: string;
  sequence_number: number;
}

export interface SessionDetailResponse {
  session: SessionSummary;
  messages: ChatMessageRecord[];
}

File: apps/api/app/schemas/sessions.py
"""
Request/response schemas for chat sessions and messages.
"""
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class SessionSummary(BaseModel):
    id: str
    title: str
    created_at: str
    updated_at: str
    message_count: int = 0


class ChatMessageRecord(BaseModel):
    id: str
    session_id: str
    role: str
    content: str = ""
    sources: List[Dict[str, Any]] = Field(default_factory=list)
    debug: Optional[Dict[str, Any]] = None
    error: Optional[str] = None
    created_at: str
    sequence_number: int


class CreateSessionRequest(BaseModel):
    title: Optional[str] = Field(None, max_length=500)


class CreateSessionResponse(BaseModel):
    session: SessionSummary


class SessionListResponse(BaseModel):
    sessions: List[SessionSummary]


class SessionDetailResponse(BaseModel):
    session: SessionSummary
    messages: List[ChatMessageRecord]


class RenameSessionRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=500)
